### PR TITLE
fix: allow initial run in session to take a while

### PR DESCRIPTION
**Pull request by itayd**
need to finess this, but this is so @efiShtain can try the approach out.
note that calling activities during initial run might not work all the time, depends on timing with the `workflow.Sleep`. We'll work it out if needed.

### CODE REVIEW

File: internal/backend/sessions/sessionworkflows/workflow.go [hunk 1]
```diff
@@ -524,20 +524,53 @@ func (w *sessionWorkflow) run(wctx workflow.Context) (prints []string, err error
...
-		return prints, err
+	type runOrErr struct {
+		run sdkservices.Run
+		err error
+	}
+
+	runDone := make(chan runOrErr)
+
+	go func() {
+		time.Sleep(20 * time.Second)
```
Comment (efiShtain) [inline internal/backend/sessions/sessionworkflows/workflow.go]: please remove this 20 seconds sleep before merging

**Review by efiShtain — APPROVED** [review]
Just don't forget to remove the 20 seconds sleep

**Comment by itayd** [comment]
> Just don't forget to remove the 20 seconds sleep

done

File: internal/backend/sessions/sessionworkflows/workflow.go [hunk 2]
```diff
@@ -524,20 +531,49 @@ func (w *sessionWorkflow) run(wctx workflow.Context) (prints []string, err error
...
-	)
-	if err != nil {
-		// TODO(ENG-130): discriminate between infra and real errors.
-		return prints, err
+	type runOrErr struct {
+		run sdkservices.Run
+		err error
+	}
+
+	runDone := make(chan runOrErr)
```
Comment (tebeka) [inline internal/backend/sessions/sessionworkflows/workflow.go:539]: Should make this buffered to avoid goroutine leak

Comment (itayd) [reply internal/backend/sessions/sessionworkflows/workflow.go:539]: https://github.com/autokitteh/autokitteh/pull/629

**Pull request merged**